ports/psoc6/main.c

```diff
@@ -114,84 +97,82 @@ int main(int argc, char **argv) {
     signal(SIGPIPE, SIG_IGN);
     #endif
 
-
-    // Initialize the device and board peripherals
-    cy_rslt_t result = cybsp_init();
-
-    // Board init failed. Stop program execution
-    if (result != CY_RSLT_SUCCESS) {
-        mp_raise_ValueError(MP_ERROR_TEXT("cybsp_init failed !\n"));
-    }
-
-
-    // Initialize retarget-io to use the debug UART port
-    result = cy_retarget_io_init(CYBSP_DEBUG_UART_TX, CYBSP_DEBUG_UART_RX, CY_RETARGET_IO_BAUDRATE);
-    error_handler(result, NULL);
-
-    // retarget-io init failed. Stop program execution
-    if (result != CY_RSLT_SUCCESS) {
-        mp_raise_ValueError(MP_ERROR_TEXT("cy_retarget_io_init failed !\n"));
-    }
-
-    is_retarget_io_initialized = true;
-
     // Initialize modules. Or to be redone after a reset and therefore to be placed next to machine_init below ?
     os_init();
     rtc_init();
     time_init();
 
-
-    #if MICROPY_PY_LWIP
-    // lwIP doesn't allow to reinitialise itself by subsequent calls to this function
-    // because the system timeout list (next_timeout) is only ever reset by BSS clearing.
-    // So for now we only init the lwIP stack once on power-up.
-    lwip_init();
-    #if LWIP_MDNS_RESPONDER
-    mdns_resp_init();
-    #endif
+    #if MICROPY_PY_NETWORK
+    cy_wcm_config_t wcm_config = { .interface = CY_WCM_INTERFACE_TYPE_STA };
+    cy_wcm_init(&wcm_config);
     #endif
 
+soft_reset:
+
+    mp_init();
 
-    result = cyhal_gpio_init(CYBSP_USER_LED, CYHAL_GPIO_DIR_OUTPUT, CYHAL_GPIO_DRIVE_STRONG, CYBSP_LED_STATE_OFF);
-    error_handler(result, NULL);
-    is_led_initialized = true;
+    // ANSI ESC sequence for clear screen. Refer to  https://stackoverflow.com/questions/517970/how-to-clear-the-interpreter-console
+    // mp_printf(&mp_plat_print, "\033[H\033[2J");
```

Comment:
This should be commented in again now that the test phase is over.

Reply:
done